chiselFrontend/src/main/scala/Chisel/Module.scala

```diff
@@ -70,14 +78,16 @@ extends HasId {
     Port(port, if (port.dir == NO_DIR) bundleDir else port.dir)
   }
 
-  private[Chisel] def setupInParent(): this.type = _parent match {
-    case Some(p) => {
-      pushCommand(DefInvalid(io.ref)) // init instance inputs
-      clock := override_clock.getOrElse(p.clock)
-      reset := override_reset.getOrElse(p.reset)
-      this
+  private[Chisel] def setupInParent(implicit sourceInfo: SourceInfo): this.type = {
```

Comment:
source information needs to be passed to DefInvalid, and even though it's guaranteed to be a UnlocatableSourceInfo (documented in the caller, basically it prevents module deduplication otherwise), I don't think it's appropriate to hardcode in UnlocatableSourceInfo in the DefInvalid for readability reasons. I guess it doesn't have to be implicit, though it helps keep things consistent.